src/main/kotlin/com/criptext/mail/BaseActivity.kt

```diff
@@ -191,12 +195,20 @@ abstract class BaseActivity: AppCompatActivity(), IHostActivity {
         cachedModels[params.activityClass] = newSceneModel
         startActivity(params.activityClass, deletePastIntents)
 
-        if (! keep) finish()
+        if (! keep) {
+            postDelayHandler.removeCallbacksAndMessages(null)
```

Comment:
when you call `finish()` it will call `onStop()` where you already are calling `postDelayHandler.removeCallbacksAndMessages`